internal/app.go

```diff
@@ -247,22 +247,34 @@ OPTIONS:
 }
 
 func showVersion(cacheDir, outputFormat, version string, outputWriter io.Writer) {
-	db.Init(cacheDir)
-	metadata, err := db.Config{}.GetMetadata()
-	if err != nil {
-		fmt.Fprintf(outputWriter, "unable to display current version: %s", err.Error())
+	if db.Init(cacheDir) != nil {
+		switch outputFormat {
+		case "json":
+			b, _ := json.Marshal(VersionInfo{
+				Version: version,
+			})
+			fmt.Fprintln(outputWriter, string(b))
+		default:
+			fmt.Fprintf(outputWriter, "Version: %s\n", version)
+		}
 		return
 	}
+
+	metadata, _ := db.Config{}.GetMetadata()
 	switch outputFormat {
 	case "json":
-		b, _ := json.Marshal(VersionInfo{
-			Version: version,
-			VulnerabilityDB: db.Metadata{
+		var dbMeta *db.Metadata
+		if !metadata.UpdatedAt.IsZero() && !metadata.NextUpdate.IsZero() && metadata.Version != 0 {
+			dbMeta = &db.Metadata{
 				Version:    metadata.Version,
 				Type:       metadata.Type,
 				NextUpdate: metadata.NextUpdate.UTC(),
 				UpdatedAt:  metadata.UpdatedAt.UTC(),
-			},
+			}
+		}
+		b, _ := json.Marshal(VersionInfo{
+			Version:         version,
+			VulnerabilityDB: dbMeta,
 		})
 		fmt.Fprintln(outputWriter, string(b))
```

Comment:
I feel like it a bit confusing to have two `switch`. How about the following implementation?
```
	var dbMeta *db.Metadata

	err := db.Init(cacheDir)
	if err == nil {
		metadata, _ := db.Config{}.GetMetadata()
		if !metadata.UpdatedAt.IsZero() && !metadata.NextUpdate.IsZero() && metadata.Version != 0 {
			dbMeta = &db.Metadata{
				Version:    metadata.Version,
				Type:       metadata.Type,
				NextUpdate: metadata.NextUpdate.UTC(),
				UpdatedAt:  metadata.UpdatedAt.UTC(),
			}
		}
	}

	switch outputFormat {
	case "json":
		b, _ := json.Marshal(VersionInfo{
			Version:         version,
			VulnerabilityDB: dbMeta,
		})
		fmt.Fprintln(outputWriter, string(b))
	default:
		output := fmt.Sprintf("Version: %s\n", version)
		if dbMeta != nil {
			var dbType string
			switch dbMeta.Type {
			case 0:
				dbType = "Full"
			case 1:
				dbType = "Light"
			}
			output += fmt.Sprintf(`Vulnerability DB:
  Type: %s
  Version: %d
  UpdatedAt: %s
  NextUpdate: %s
`, dbType, dbMeta.Version, dbMeta.UpdatedAt.UTC(), dbMeta.NextUpdate.UTC())
		}
		fmt.Fprintf(outputWriter, output)
	}
```

Reply:
Good point! I have DRY'd it up: https://github.com/aquasecurity/trivy/pull/443/commits/02ccd041a9230ba85b193baebd10b96c2104a7bc